src/main/java/seedu/address/model/food/Fridge.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.model.food;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.food.exceptions.FoodNotFoundException;
+
+/**
+ * A list of food items that allows repeated elements and does not allow nulls.
+ * Supports a minimal set of list operations.
+ *
+ * @see Food#isSameFood(Food)
+ */
+public class Fridge implements Iterable<Food> {
+    private final ObservableList<Food> internalList = FXCollections.observableArrayList();
+    private final ObservableList<Food> internalUnmodifiableList =
+        FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent food item as the given argument.
+     */
+    public boolean contains(Food toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameFood);
+    }
+
+    /**
+     * Adds a food item to the list.
+     */
+    public void add(Food toAdd) {
+        requireNonNull(toAdd);
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Replaces the food item {@code target} in the list with {@code editedFood}.
+     * {@code target} must exist in the list.
+     */
+    public void setFood(Food target, Food editedFood) {
```

Comment:
This method only makes sense if we have unique objects in the list, maybe you could change the `target` to the `Index` of the object in the list, or require all objects in the `fridge` to be unique.

Reply:
Fixed